Setting __STDC_VERSION__ to a nonsense value won't lead to a successful outcome.

--- a/numpy/core/meson.build	Mon Feb  5 21:17:48 2024
+++ b/numpy/core/meson.build	Sun Jun  2 20:54:27 2024
@@ -632,9 +632,7 @@
 ]
 
 # Same as NPY_CXX_FLAGS (TODO: extend for what ccompiler_opt adds)
-cpp_args_common = c_args_common + [
-  '-D__STDC_VERSION__=0',  # for compatibility with C headers
-]
+cpp_args_common = c_args_common
 if cc.get_argument_syntax() != 'msvc'
   cpp_args_common += [
     '-fno-exceptions',  # no exception support
